### PR TITLE
reconcile pull secret for fuse and 3scale

### PR DESCRIPTION
**Pull request by philbrookes**
- Prepare for using GA version of 3scale operator
- Abstract docker pull secret reconciliation
- Update unit tests

### CODE REVIEW

File: deploy/operator.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ spec:
       containers:
         - name: integreatly-operator
           # Replace this with the built image name
-          image: quay.io/integreatly/integreatly-operator:v1.8.1
+          image: quay.io/integreatly/integreatly-operator:v1.8.2
```
Comment (austincunningham) [inline deploy/operator.yaml]: The manifest is still pointing at v1.8.0 , is there a version gone missing somewhere https://github.com/integr8ly/manifests/blob/master/Makefile#L2

Comment (philbrookes) [reply deploy/operator.yaml]: maybe, I'll catch it up when I PR

**Comment by philbrookes** [comment]
@austincunningham for verification you can check that the sample secret is copied into the 3scale namespace and the fuse namespace

File: deploy/operator.yaml (hunk 1 above)
Comment (austincunningham) [reply deploy/operator.yaml]: :+1:

**Pull request merged**